File: src/Modules/Procurement/ErpClink.Modules.Procurement.Application/Suppliers/Models/SupplierModels.cs
namespace ErpClink.Modules.Procurement.Application.Suppliers.Models;

public sealed record CreateSupplierRequest(
    string Name,
    string? ContactName,
    string? Phone,
    string? Email,
    string? Address,
    string? Notes);

public sealed record UpdateSupplierRequest(
    string Name,
    string? ContactName,
    string? Phone,
    string? Email,
    string? Address,
    string? Notes,
    byte[]? RowVersion);

public sealed record SearchSuppliersRequest(
    string? Search,
    bool? IsActive,
    int Page = 1,
    int PageSize = 20);

public sealed record SupplierDto(
    Guid Id,
    Guid OrganizationId,
    string SupplierCode,
    string Name,
    string? ContactName,
    string? Phone,
    string? Email,
    string? Address,
    string? Notes,
    bool IsActive,
    DateTime CreatedAtUtc,
    string? CreatedBy,
    DateTime? UpdatedAtUtc,
    string? UpdatedBy,
    byte[] RowVersion);

public sealed record PagedSuppliersResult(
    IReadOnlyList<SupplierDto> Items,
    int TotalCount,
    int Page,
    int PageSize);

File: frontend/erpclink-web/src/app/features/procurement/models/supplier.models.ts
export interface SupplierDto {
  id: string;
  organizationId: string;
  supplierCode: string;
  name: string;
  contactName?: string | null;
  phone?: string | null;
  email?: string | null;
  address?: string | null;
  notes?: string | null;
  isActive: boolean;
  createdAtUtc: string;
  createdBy?: string | null;
  updatedAtUtc?: string | null;
  updatedBy?: string | null;
  rowVersion: string;
}

export interface PagedSuppliersResult {
  items: SupplierDto[];
  totalCount: number;
  page: number;
  pageSize: number;
}

export interface CreateSupplierRequest {
  name: string;
  contactName?: string | null;
  phone?: string | null;
  email?: string | null;
  address?: string | null;
  notes?: string | null;
}

export interface UpdateSupplierRequest extends CreateSupplierRequest {
  rowVersion?: string | null;
}
